File: code wars/7kyu/Looking for a benefactor/main.py
import math

# Taking the equation :  (sum + x) / (len + 1) = navg
# then x should be equal to (navg * (len + 1)) - sum

def new_avg(arr, newavg):
    if  newavg * (len(arr) + 1) > sum(arr):
        return math.ceil(newavg * (len(arr) + 1) - sum(arr))
    raise Error("Error expected")

File: code wars/7kyu/Looking for a benefactor/main.ts
function newAvg(arr: number[], newavg: number): number {
    let res: number = (newavg * (arr.length + 1)) - arr.reduce((a, b) => a + b, 0)
    if (res <= 0) throw "Expected New Average is too low"
    return Math.ceil(res)
  }